envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -222,6 +222,19 @@ message HttpConnectionManager {
   // downstream. If this is false or not set, Envoy will instead strip the
   // "Expect: 100-continue" header, and send a "100 Continue" response itself.
   bool proxy_100_continue = 18;
+
+  // If
+  // :ref:`use_remote_address
+  // <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.use_remote_address>`
+  // is true and remote_address_ipv4_mapped is true and the remote address is an IPv4 address, the
+  // address will be mapped to IPv6 before it is appended to x-forwarded-for. This is useful for
+  // testing compatibility of upstream services that parse the header value. For example, 50.0.0.1
+  // is represented as ::FFFF:50.0.0.1. See `IPv4-Mapped IPv6 Addresses
+  // <https://tools.ietf.org/html/rfc4291#section-2.5.5.2>`_ for details. This will also affect the
+  // :ref:`config_http_conn_man_headers_x-envoy-external-address` header. See
+  // :ref:`http_connection_manager.remote_address_ipv4_mapped_ipv6
+  // <config_http_conn_man_runtime_remote_address_ipv4_mapped_ipv6>` for runtime control.
+  bool remote_address_ipv4_mapped_ipv6 = 19;
```

Comment:
What about "convert_remote_address_to_ipv4_mapped_ipv6"? It's long, but maybe easier to understand what the field does? Also, follow up on @htuch comment above, should we make this a Percent if we are going to allow a runtime percentage? Not sure if it's worth it but throwing it out there.

Reply:
Yeah, I don't mind making it bool for config and percentage for runtime, I actually blanked on that being runtime. SGTM.

Reply:
sure i can go with that name. it's again confusing because these are called ipv4 mapped ipv6 addresses. so technically the correct name would be `convert_ipv4_remote_address_to_ipv4_mapped_ipv6`. in fact i think i'll go with that

Reply:
SGTM